koji_wrapper/factory.py

```diff
@@ -0,0 +1,48 @@
+
+from os import path
+
+import koji
+from koji_wrapper.exceptions import UnknownAuthMethod
+
+
+class KojiWrapperFactory(object):
+
+    @classmethod
+    def load_profile(cls, profile='koji', user_config=None):
+        try:
+            result = koji.read_config(profile, user_config=user_config)
+            for k, v in result.items():
+                result[k] = path.expanduser(v) if type(v) is str else v
+        except koji.ConfigurationError as err:
+            raise err
+
+        return result
+
+    @classmethod
+    def open_session(cls, profile=None, user_options=None,
```

Comment:
I am not sure why this needs to be a class method.  Why not just have the standard init call this and return a ClientSession object?  Then the user could call it right in their KojiWrapper instantiate, like this:
``
kw = KojiWrapper(session=KojiSessionFactory(profile='foo'))
``

Reply:
Reasoning I was using is that opening a koji.ClientSession is not a one size fits all option, and you may want to re-use an existing session or have other ways of creating it that would work for you but might not work for all ways.  So a Factory object would be the approach I would use.  This is the initial attempt at the factory object but I could see more options or other ways of doing this being exposed.  But it allows us to wrap the nitty gritty of what has to be done to open a Session object that is usable.

and yes the usage of kw = KojiWrapper(session=KojiSessionFactory(profile='foo')) is most likely what would be used.